spec/unit/controllers/v3/application_controller_spec.rb

```diff
@@ -235,17 +248,47 @@ def not_found
     end
 
     it 'asks for #can_read_from_space? on behalf of the current user' do
-      permissions = instance_double(VCAP::CloudController::Permissions, can_read_from_space?: true)
+      permissions = instance_double(
```

Comment:
Agreed. The reason for keeping things in the controller **for now** was to minimize the code change delta as small. The hope was also to get this merged a couple weeks ago so that we could refactor based on master.